### DISCUSSION TITLE
XAML - Show / hide on specific platform

### DISCUSSION THREAD

**Discussion by Symbai** [dotnet/maui#13442] [Q&A] [answered]
Is there an easy way I can show / hide xaml content like a button, border etc. only on for example iOS? Like in code behind I can use pre processor directives like

```
#if ANDROID
...
#else
...
#end
```

Is there something like that for XAML code?

**Comment by acaliaro** [comment, marked as answer]
I think there is the OnPlatform directive in xaml and you can use it also in csharp

**Reply by Symbai** [reply]
But it only works for single properties like:

```XAML
<Button IsVisible={OnPlatform Android=True, IOS=False}/>

```
I was more looking for something like:

```XAML
<OnPlatform Android>

  <Button/>
  <Border/>
  <Listview/>
  <Whatever/>

</OnPlatform>
```

**Reply by acaliaro** [reply]
Set the layout property (for example set stacklayout IsVisible )

**Reply by Symbai** [reply]
I fear that its still there on other platforms, but just invisible. Which means that it costs unnecessary performance. But seems like there is no other solution.
